src/Diagnostics.RuntimeHost/Services/AssemblyCacheService.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+namespace Diagnostics.RuntimeHost.Services
+{
+    public interface IAssemblyCacheService
+    {
+        /// <summary>
+        /// Adds a given <paramref name="assemblyName"/> to the cache. If a limit of <see cref="MaxQueueSize"/> is reached, oldest assembly is removed
+        /// </summary>
+        /// <param name="assemblyName">Full Qualified name of the DLL to cache</param>
+        /// <param name="assemblyDll">DLL to add to cache</param>
+        void AddAssemblyToCache(string assemblyName, Assembly assemblyDll);
+
+        /// <summary>
+        /// Checks if a <paramref name="assemblyName"/> is loaded in cache and returns <paramref name="loadedAssembly"/>
+        /// </summary>
+        bool IsAssemblyLoaded(string assemblyName, out Assembly loadedAssembly);
+    }
+
+    public class AssemblyCacheService: IAssemblyCacheService
+    {
+        /// <summary>
+        /// Queue to maintain top <see cref="MaxQueueSize"/> assemblies
+        /// </summary>
+        Queue<string> AssemblyQueue;
```

Comment:
Use a ConcurrentQueue instead of a normal Queue as the later is not thread-safe.

Reply:
Also qualify the members as private